egressgw: policy: ensure egressGateway field is not nil

jibi opened the pull request.

jibi committed:
egressgw: policy: ensure egressGateway field is not nil

Reported-by: Ralph Bankston <ralph.bankston@isovalent.com>
Signed-off-by: Gilberto Bertin <jibi@cilium.io>

julianwiedmann commented on pkg/egressgateway/policy.go:
```diff
@@ -261,6 +261,10 @@ func ParseCEGP(cegp *v2.CiliumEgressGatewayPolicy) (*PolicyConfig, error) {
+	if egressGateway == nil {
+		return nil, fmt.Errorf("CiliumEgressGatewayPolicy's egressGateway can't be empty")
+	}
+
```
drive-by - in some of the error paths further below, we're already producing a formatted error with the resource name. Let's clean that up at some point, it would help users to understand which CEGP is the culprit.

jibi replied:
good shout, after the switch to resource[T] we are not actually logging any of these errors :grimacing:  so I'll refactor this function a bit and make the caller log the error together with the CEGP name

julianwiedmann reviewed (approved):
thanks!

jibi committed:
egressgw: policy: emit a warning for ParseCEGP errors

after the switch to the Resource[T] mechanism to handle
CiliumEgressGatewayPolicy events, we don't log errors from ParseCEGP()
anymore. This commit reintroduces the warnings.

Signed-off-by: Gilberto Bertin <jibi@cilium.io>
